src/test/java/org/dataloader/DataLoaderMappedObserverBatchLoaderTest.java

```diff
@@ -0,0 +1,106 @@
+        await().untilAtomic(success, is(true));
+    }
+
+    @Test
+    public void should_Support_loading_multiple_keys_in_one_call() {
+        AtomicBoolean success = new AtomicBoolean();
+        DataLoader<Integer, Integer> identityLoader = mkDataLoader(keysAsValues(), DataLoaderOptions.newOptions());
+
+        CompletionStage<List<Integer>> futureAll = identityLoader.loadMany(asList(1, 2));
+        futureAll.thenAccept(promisedValues -> {
+            assertThat(promisedValues.size(), is(2));
+            success.set(true);
+        });
+        identityLoader.dispatch();
+        await().untilAtomic(success, is(true));
+        assertThat(futureAll.toCompletableFuture().join(), equalTo(asList(1, 2)));
+    }
+
+    @Test
+    public void simple_dataloader() {
+        DataLoader<String, String> loader = mkDataLoader(keysAsValues(), DataLoaderOptions.newOptions());
+
+        loader.load("A");
+        loader.load("B");
+        loader.loadMany(asList("C", "D"));
+
+        List<String> results = loader.dispatchAndJoin();
+
+        assertThat(results.size(), equalTo(4));
+        assertThat(results, equalTo(asList("A", "B", "C", "D")));
+    }
+
+    @Test
+    public void should_observer_batch_multiple_requests() throws ExecutionException, InterruptedException {
+        DataLoader<Integer, Integer> identityLoader = mkDataLoader(keysAsValues(), new DataLoaderOptions());
+
+        CompletableFuture<Integer> future1 = identityLoader.load(1);
+        CompletableFuture<Integer> future2 = identityLoader.load(2);
+        identityLoader.dispatch();
+
+        await().until(() -> future1.isDone() && future2.isDone());
+        assertThat(future1.get(), equalTo(1));
+        assertThat(future2.get(), equalTo(2));
+    }
+
+    // A simple wrapper class intended as a proof external libraries can leverage this.
+    private static class Publisher<K, V> {
+
+        private final MappedBatchObserver<K, V> delegate;
+        private Publisher(MappedBatchObserver<K, V> delegate) { this.delegate = delegate; }
+        void onNext(Map.Entry<K, V> entry) { delegate.onNext(entry.getKey(), entry.getValue()); }
+        void onCompleted() { delegate.onCompleted(); }
+        void onError(Throwable e) { delegate.onError(e); }
+        // Mock 'subscribe' methods to simulate what would happen in the real thing.
+        void subscribe(Map<K, V> valueByKey) {
+            valueByKey.entrySet().forEach(this::onNext);
+            this.onCompleted();
+        }
+        void subscribe(Map<K, V> valueByKey, Throwable e) {
+            valueByKey.entrySet().forEach(this::onNext);
+            this.onError(e);
+        }
+    }
+
+    private static <K> MappedObserverBatchLoader<K, K> keysAsValues() {
+        return (keys, observer) -> {
+            Publisher<K, K> publisher = new Publisher<>(observer);
+            Map<K, K> valueByKey =  keys.stream().collect(toMap(identity(), identity()));
+            publisher.subscribe(valueByKey);
```

Comment:
I see this idea of subscription as a key aspect to be addressed.

Imagine some one had a reactive real Publisher of events.  At what point is a subscription made?

Do people have to write this own wrappers so that they can move between a Reactor stream of items and this `ObserverBatchLoader / MappedObserverBatchLoader` say?  

This could be tedious.  In theory this library COULD become dependent on reactive streams like graphql-java is

```groovy
    def reactiveStreamsVersion = '1.0.3'
    api 'org.reactivestreams:reactive-streams:' + reactiveStreamsVersion
```

This would allow you to have interfaces that can take a official `Publisher` of events and you could write the `Subscription` interfacing code that made your actual `ObserverBatchLoader 
 / MappedObserverBatchLoader`

I think if we allowed this PR then I would have this extra dependency - not to a implementation like Reactor - but to the raw streams api

Reply:
Imagine adator code like

```
return (keys, observer) -> {
     Publisher reactiveImplPublisher = createReactivePublisherFromKeys(keys);
     ReactiveAdapater.subscribeTo(reactiveImplPublisher, observer);
}
```

This would allow you to call put to a real reactive system and then adapt the publisher back to this observer via this helper code we supply.  Rather than writing a subscription youself that ends up fowarding everything to the `observer`
```

Reply:
Done, these now are based on `reactive-stream`'s `Subscriber`s.